mongo/integration/unified/matches.go

```diff
@@ -55,7 +55,11 @@ func verifyValuesMatchInner(ctx context.Context, expected, actual bson.RawValue)
 
 		actualDoc, ok := actual.DocumentOK()
 		if !ok {
-			return newMatchingError(keyPath, "expected value to be a document but got a %s", actual.Type)
+			// If actual value isn't empty, return error.
+			if len(actual.Value) != 0 {
+				return newMatchingError(keyPath, "expected value to be a document but got a %s", actual.Type)
+			}
+			actualDoc = bson.Raw{}
```

Comment:
If `actual` is just empty (just `bson.RawValue{}`), as is the case with redacted commands, `DocumentOK()` will return `ok = false` because there are not enough bytes to read a document from `bson.RawValue{}`. I added a workaround here for empty, actual documents. Let me know if you think this is odd/doesn't belong here.

Reply:
It is initially unclear why `DocumentOK` returning "not ok" should be OK. Optional: Invert the logic here to check for an empty document first, then only use `DocumentOK()` if it's not empty.

E.g.:
```go
var actualDoc bson.Raw
// If actual is empty, as is the case with redacted commands, DocumentOK() will return "ok = false"
// because there are not enough bytes to read a document from bson.RawValue{}. In that case, just
// use an empty bson.Raw document.
if len(actual.Value) == 0 {
	actualDoc = bson.Raw{}
} else {
	var ok bool
	actualDoc, ok = actual.DocumentOK()
	if !ok {
		return newMatchingError(keyPath, "expected value to be a document but got a %s", actual.Type)
	}
}
```

Reply:
This may also require checking that `actual.Type` is still `bsontype.EmbeddedDocument`. Otherwise I would worry this could change the matching behavior for: `expected = {a: {}}, actual = {a: null}` since a [BSON null](https://bsonspec.org/spec.html) has no payload.

However, based on https://github.com/mongodb/mongo-go-driver/blob/825803382d301b8e1bc975f8c292f5317db21e38/x/mongo/driver/operation.go#L1434:L1455
it looks like this is particular to how `CommandStartedEvent` and `CommandSucceededEvent` expose the `Command` field when redacting.

If they are redacted, the `Command` is left as an empty slice.

If so, another alternative could be to leave the match logic, and instead change the callers in  event_verification.go (https://github.com/mongodb/mongo-go-driver/blob/825803382d301b8e1bc975f8c292f5317db21e38/mongo/integration/unified/event_verification.go#L167:L173) to use an empty document if the `Command` is an empty slice.

Reply:
Ah good point! I've left the match logic as is and changed the callers in `event_verification` to use an empty document in case of an empty `actual.Command`. I had to hardcode a `bson.RawValue` representing an empty document, so let me know if it looks strange. 

Also updated the comments to be more clear as @matthewdale had written above.